test/index.js

```diff
@@ -56,107 +67,201 @@ const testUtils = {
 };
 
 const envDB = process.env.DB;
-const testDbs = (envDB && [envDB]) || ['postgres'];
+const testDb = envDB || 'postgres';
+
+const knexConfig = KnexConfigs.find((conf) => conf.client === testDb);
+
+if (!knexConfig) {
+    throw new Error(`Unsupported db "${testDb}"`);
+}
 
 describe('SchwiftyMigration', () => {
 
-    let initSessions = [];
+    const makeSession = (cb) => {
 
-    const getSessions = () => {
+        const session = new TestSession({ options: { knexConfig } },
+            (err) => {
 
-        const sessions = [];
+                setOptionsForAfter(session);
+                cb(err, session);
+            });
+    };
 
-        const promises = KnexConfigs.filter((knexConfig) => {
+    const failKnexWith = (knex, toErrorOn, errMsg, afterTries) => {
 
-            // Only test the dbs specified in the `testDbs` array
-            return testDbs.indexOf(knexConfig.client) !== -1;
-        })
-            .map((knexConfig) => {
+        // Grabbed this technique from https://github.com/tgriesser/knex/blob/2e1a459a9e740f24b9a4647bd4da427854e551dd/test/integration/logger.js#L89-L108
 
-                return new Promise((resolve, reject) => {
+        afterTries = afterTries || 1;
 
-                    // Create all the test sessions
+        const originalQb = knex.queryBuilder;
+        knex.queryBuilder = () => {
 
-                    sessions.push(new TestSession({
-                        options: { knexConfig },
-                        next: () => {
+            const qb = originalQb.apply(this, arguments);
+            const origToErrorFunc = qb[toErrorOn].bind(qb);
 
-                            resolve(...sessions);
-                        }
-                    }));
-                });
-            });
+            qb[toErrorOn] = (...args) => {
+
+                if (--afterTries === 0) {
+                    return Promise.reject(new Error(errMsg));
```

Comment:
You probably want to ensure you return knex/objection's `Promise`s from mocks to ensure `asCallback()` is available.